### PR TITLE
Initial documentation

### PR DESCRIPTION
**Pull request by NicolasT**
Add some initial documentation, based on Sphinx.

### CODE REVIEW

**Comment by NicolasT** [comment]
Would :heart: approval from @thomasdanan once completed the Quickstart guide.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+For the really impatient, there's a [quickstart] available.
 
-## Test config
-If you want  to test metal-k8s you just need one simple VM running ubuntu
- 16.04
+[quickstart]: https://metal-k8s.readthedocs.io/en/latest/usage/quickstart.html
 
-## Production Config
+## Documentation
+The project documentation is available at https://metal-k8s.readthedocs.io.
```
Comment (LaureVergeron) [inline README.md]: Shouldn't this be a link?

File: README.md [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+Sources can be found in the `docs` folder of this repository. Pull-requests
+welcome!
+
+## Hardware Requirements
+
+### Test config
+If you want  to test metal-k8s you just need one simple VM running CentOS 7.4.
```
Comment (LaureVergeron) [inline README.md]: double space

File: README.md [hunk 3]
```diff
@@ -17,56 +26,5 @@ For production-ready environment, you need:
...
-## Usage
-You also need `kubectl` and `helm` binary to play with your cluster from your
-local computer.
-
-Please refer to the official documentation to install those tools:
-
-- https://kubernetes.io/docs/tasks/tools/install-kubectl/
-- https://docs.helm.sh/using_helm/#installing-helm
+## Software Requirements
+To get started, all you need is `python` and `make` on a Unix-like system.
```
Comment (LaureVergeron) [inline README.md]: which versions?

Comment (NicolasT) [reply README.md]: Doesn't matter, which is exactly the point.

File: README.md (hunk 1 above)
Comment (NicolasT) [reply README.md]: It is.

File: README.md (hunk 3 above)
Comment (NicolasT) [reply README.md]: @wabernatScality Feedback welcome at any point in time as far as I'm concerned :smile: Though maybe a bit too early now, I'll add more content first. Thanks for the :+1:!

File: README.md (hunk 2 above)
Comment (NicolasT) [reply README.md]: Fixed

**Comment by NicolasT** [comment]
Change since previous reviews: addition of the *Cluster Services* document, see https://github.com/scality/metal-k8s/blob/docs/docs/architecture/cluster-services.rst

**Comment by NicolasT** [comment]
Agree we should try to get patterns set up in Kibana by default, though that's out of scope for this PR :) I guess something should be feasible through a Helm `init job`.

**Pull request merged**